### PR TITLE
[task] remove all hardcoded links to devdocs.magento.com with {{ site.baseurl }}

### PR DESCRIPTION
**Pull request by mhauri**
## This PR is a:
- [ ] New topic
- [ ] Content fix or rewrite
- [x] Bug fix or improvement
 

## Summary
 
When this pull request is merged, it will replace all hardcoded links starting with `https://devdocs.magento.com` with the variable `{{ site.baseurl }}`.
 

## Warning
**This PR affects a huge amount of files**, the benefit on the other hand is that if you run `jekyll serve` locally it will prevent jumping to the live site when clicking on a link.

### CODE REVIEW

File: guides/v2.1/rest/apis.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ title: REST APIs (Swagger)
 menu_title: REST APIs (Swagger)
 version: 2.1
 menu_order: 2
-externalURL: http://devdocs.magento.com/swagger/index_21.html
+externalURL: {{ site.baseurl }}/swagger/index_21.html
```
Comment (hguthrie) [inline guides/v2.1/rest/apis.md]: @jeff-matthews This breaks the bundle serve for me. Can you double check?

Comment (mhauri) [reply guides/v2.1/rest/apis.md]: @hguthrie you're right, noticed it as well as I tested it again. Will create a PR to fix.

**Comment by keharper** [comment]
I'll fix this.

**Comment by jeff-matthews** [comment]
Thanks @hguthrie for flagging this and thanks @keharper for fixing it. My apologies for not testing more thoroughly before merging.

**Pull request merged**